diff --git a/client/apollo/js/main.js b/client/apollo/js/main.js
index 6b5f4b200..78502f221 100644
--- a/client/apollo/js/main.js
+++ b/client/apollo/js/main.js
@@ -357,18 +357,24 @@ return declare( [JBPlugin, HelpMixin],
     initSearchMenu: function()  {
         if (! this.searchMenuInitialized) {
             var webapollo = this;
-            this.browser.addGlobalMenuItem( 'tools',
-                                            new dijitMenuItem(
-                                                {
-                                                    id: 'menubar_apollo_seqsearch',
-                                                    label: "Search sequence",
-                                                    onClick: function() {
-                                                        webapollo.getAnnotTrack().searchSequence();
-                                                    }
-                                                }) );
-            if(!dijitRegistry.byId("dropdownmenu_tools")){
-                this.browser.renderGlobalMenu( 'tools', {text: 'Tools'}, this.browser.menuBar );
-            }
+            this.browser.afterMilestone('initView', function() {
+                var button = new dijitMenuItem(
+                                {
+                                    id: 'menubar_apollo_seqsearch',
+                                    label: "Search sequence",
+                                    onClick: function() {
+                                        webapollo.getAnnotTrack().searchSequence();
+                                    }
+                                }) ;
+                this.browser.addGlobalMenuItem( 'tools', button);
+                var renderedMenu = dijitRegistry.byId("dropdownmenu_tools");
+                if(!renderedMenu){
+                    this.browser.renderGlobalMenu( 'tools', {text: 'Tools'}, this.browser.menuBar );
+                }
+                else {
+                    renderedMenu.addChild(button);
+                }
+            }, this );
 
         }
 
