diff --git a/grails-app/conf/org/bbop/apollo/SecurityFilters.groovy b/grails-app/conf/org/bbop/apollo/SecurityFilters.groovy
index 669d962b4..7b33b7c56 100644
--- a/grails-app/conf/org/bbop/apollo/SecurityFilters.groovy
+++ b/grails-app/conf/org/bbop/apollo/SecurityFilters.groovy
@@ -80,7 +80,7 @@ class SecurityFilters {
 //                                }
                                 targetUri = targetUri + paramString
                                 if (paramString.contains("http://") || paramString.contains("https://") || paramString.contains("ftp://")) {
-                                    redirect(uri: "${request.contextPath}/auth/login?targetUri=${targetUri}")
+                                    redirect(uri: System.getenv("APOLLO_BASE_HOST") + System.getenv("APOLLO_PATH_PREFIX") + "/auth/login?targetUri=${targetUri}")
                                 }
                                 else {
                                     redirect(uri: "/auth/login?targetUri=${targetUri}")
diff --git a/grails-app/controllers/org/bbop/apollo/AnnotatorController.groovy b/grails-app/controllers/org/bbop/apollo/AnnotatorController.groovy
index 2f822916d..dda3e1541 100644
--- a/grails-app/controllers/org/bbop/apollo/AnnotatorController.groovy
+++ b/grails-app/controllers/org/bbop/apollo/AnnotatorController.groovy
@@ -164,7 +164,7 @@ class AnnotatorController {
         }
         if (queryParamString.contains("http://") || queryParamString.contains("https://") ||
                 queryParamString.contains("ftp://")) {
-            redirect uri: "${request.contextPath}/annotator/index?clientToken=" + clientToken + queryParamString
+            redirect uri: System.getenv("APOLLO_BASE_HOST") + System.getenv("APOLLO_PATH_PREFIX") + "/annotator/index?clientToken=" + clientToken + queryParamString
         } else {
             redirect uri: "/annotator/index?clientToken=" + clientToken + queryParamString
         }
