diff --git a/docs/web_services/examples/shell/listStatus.sh b/docs/web_services/examples/shell/listStatus.sh
new file mode 100755
index 000000000..633fb8b8e
--- /dev/null
+++ b/docs/web_services/examples/shell/listStatus.sh
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+url=$1
+username=$2
+password=$3
+organism=$4
+trackName=$5
+uniqueName=$6
+status=$7
+
+usage() {
+    echo "Sample script for listing statuses via web services"
+    echo "Usage:    ./listStatus.sh <complete_apollo_URL> <username> <password> "
+    echo "Example:  ./listStatus.sh http://localhost:8080/apollo demo demopass"
+    echo ""
+}
+
+if [[ ! -n "$url" || ! -n "$username" || ! -n "$password"  ]]; then
+    usage
+    exit
+fi
+
+echo curl -i -H 'Content-type: application/json' -X POST ${url}/availableStatus/showStatus -d "{'username': '${username}', 'password': '${password}'}"
+curl -i -H 'Content-type: application/json' -X POST ${url}/availableStatus/showStatus -d "{'username': '${username}', 'password': '${password}'}"
diff --git a/grails-app/controllers/org/bbop/apollo/AvailableStatusController.groovy b/grails-app/controllers/org/bbop/apollo/AvailableStatusController.groovy
index 4af634947..805595fc0 100644
--- a/grails-app/controllers/org/bbop/apollo/AvailableStatusController.groovy
+++ b/grails-app/controllers/org/bbop/apollo/AvailableStatusController.groovy
@@ -31,8 +31,6 @@ class AvailableStatusController {
     }
 
     def index(Integer max) {
-//        params.max = Math.min(max ?: 10, 100)
-//        respond AvailableStatus.list(params), model:[availableStatusInstanceCount: AvailableStatus.count()]
         params.max = Math.min(max ?: 10, 100)
         def availableStatuss = AvailableStatus.list(params)
         def organismFilterMap = [:]
diff --git a/grails-app/services/org/bbop/apollo/PermissionService.groovy b/grails-app/services/org/bbop/apollo/PermissionService.groovy
index df9154581..e2ab19b15 100644
--- a/grails-app/services/org/bbop/apollo/PermissionService.groovy
+++ b/grails-app/services/org/bbop/apollo/PermissionService.groovy
@@ -345,6 +345,11 @@ class PermissionService {
             Session session = SecurityUtils.subject.getSession(false)
             if (session) {
                 Map<String, Integer> permissions = (Map<String, Integer>) session.getAttribute(FeatureStringEnum.PERMISSIONS.getValue())
+                // permissions not always on session if they come through a web-service, see #1759
+                if(!permissions){
+                    User user = User.findByUsername(SecurityUtils.subject.principal.toString())
+                    permissions = getPermissionsForUser(user)
+                }
                 if (permissions) {
                     Integer permission = permissions.get(SecurityUtils.subject.principal)
                     PermissionEnum sessionPermissionsEnum = isAdmin() ? PermissionEnum.ADMINISTRATE : PermissionEnum.getValueForOldInteger(permission)
